src/components/Error/ErrorView.test.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { Router } from 'react-router-dom';
+import { act, render, RenderResult, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { createMemoryHistory } from 'history';
+
+import {
+  ErrorView,
+  ErrorViewProps,
+  internalServerErrorText,
+  labels,
+  TIME_TO_RELOAD_SEC,
+} from './ErrorView';
+
+const { reload } = window.location;
+
+const history = createMemoryHistory({ initialEntries: ['/'] });
+
+beforeEach(() => {
+  jest.useFakeTimers();
+  Object.defineProperty(global, 'location', {
+    writable: true,
+    value: { reload: jest.fn() },
+  });
+});
+
+afterEach(() => {
+  global.location.reload = reload;
+  jest.useRealTimers();
+});
+
+function renderComponent(props: Partial<ErrorViewProps> = {}): RenderResult {
+  return render(
+    <Router history={history}>
+      <ErrorView code={400} message="validation-error" {...props} />
+    </Router>,
+  );
+}
+
+describe('ErrorView', () => {
+  test('корректно рендерится', () => {
+    expect(renderComponent).not.toThrow();
+  });
+
+  test('при 500 ошибке страница перезагружается через TIME_TO_RELOAD_SEC секунд', () => {
+    renderComponent({ code: 500 });
+
+    act(() => {
+      jest.advanceTimersByTime(TIME_TO_RELOAD_SEC * 1000);
+    });
+
+    expect(global.location.reload).toHaveBeenCalled();
+  });
+
+  test(`при 500 ошибке рендерится кнопка "${labels.reloadButton}"`, () => {
+    renderComponent({ code: 500 });
+
+    expect(screen.queryByLabelText(labels.reloadButton)).toBeInTheDocument();
+  });
+
+  test(`рендерится кнопка "${labels.projectButton}"`, () => {
+    renderComponent({ code: 404 });
+
+    expect(screen.queryByLabelText(labels.projectButton)).toBeInTheDocument();
+  });
+
+  test(`при нажатии на кнопку "${labels.reloadButton}" происходит обновление страницы`, () => {
+    renderComponent({ code: 500 });
+
+    const button = screen.queryByLabelText(labels.reloadButton);
```

Comment:
```suggestion
    const button = screen.getByRole('button', { name: 'Попробовать снова' });
```
Это просто как предложение. Тут вопрос ко всем. Что думаете? Мне кажется, что так проще читать, и в целом должно быть надежней

Reply:
искать по роли?

Reply:
ага, и по тексту интерактивного элемента